opm/parser/eclipse/EclipseState/Grid/NNC.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+  Copyright 2015 IRIS
+  
+  This file is part of the Open Porous Media project (OPM).
+  
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "NNC.hpp"
+#include <array>
+
+namespace Opm
+{
+    NNC::NNC(Opm::DeckConstPtr deck, EclipseGridConstPtr eclipseGrid)
+    {
+        if (deck->hasKeyword("NNC")) {
+            const std::vector<DeckKeywordConstPtr>& nncs = deck->getKeywordList("NNC");
+            for (size_t idx_nnc = 0; idx_nnc<nncs.size(); ++idx_nnc) {
+                Opm::DeckKeywordConstPtr nnc = nncs[idx_nnc];
+                size_t numNNC = nnc->size();
+                nnc1_.reserve(numNNC);
```

Comment:
> So the case is actually that we do not know the size beforehand?

Well, yeah, kind of.  The total number of `NNC`s is the sum of the `->size()` of each `NNC` keyword.  It is entirely trivial to precompute that number in a separate loop, e.g.,
```C++
std::vector<double>::size_type numNNC = 0;
for (auto& b = nncs.begin(), e = nncs.end(); b != e; ++b) {
    numNNC += b->size();
}
```
and then use `numNNC` in the `.reserve()` call.